### ISSUE TITLE
Native JSI functions

### ISSUE THREAD

**Issue by gillpeacegood** [mrousavy/react-native-multithreading#24]
```
 ("host functions", aka functions that print function f() { [native code] } when you call .toString() on them) can be called synchronously (e.g. functions from react-native-mmkv)
 ```
 
I am trying to do this but the host function doesn't seem to get called. A break point in the host function in xcode does not get hit

I was wondering if you had any example code I could refer to?

My code looks like

```
// database previously opened okay (synchronously, via JSI)
const quick_database_name = 'QuickTest.db';

// global variable exposed by JSI
declare var sqlite;


const doQuickSql = async (statement: string, params?: Array<any>) : Promise<[ResultSet]> => {
  let _sqlite = sqlite;

  const result = await spawnThread( () => {
    'worklet'
    console.log(_sqlite, quick_database_name, statement, params);
    const result = _sqlite.executeSql(quick_database_name, statement, params);
    console.log(result);
    return result;
  });
  console.log(result);
  return [transformResults(result)];
}

```

With the console log indicating that the function exists but a break point in the JSI implementation is never hit in XCode and the result undefined

The same function works okay if it is called synchronously

**Comment by mrousavy** [comment]
I found out that Reanimated needs to know about the host function - so try [following the guide in the VisionCamera frame processor docs](https://cuvent.github.io/react-native-vision-camera/docs/guides/frame-processors-plugins-final), where you need to add the host-function's name to the babel plugin and try again. You will need to use reanimated from master though, at least until [my PR](https://github.com/software-mansion/react-native-reanimated/commit/11250a027fbea0abc719a2af2ffcd0322e145395) has been shipped in a release.

**Comment by gillpeacegood** [comment]
I have tried all that and still it doesn't work.

Would be great if you happen to have example code using MMKV that works

**Comment by mrousavy** [comment]
I just noticed that [in VisionCamera I also install the host functions into the VisionCamera Frame Processor Runtime](https://github.com/cuvent/react-native-vision-camera/blob/160ec779404be276f4953d3d760bc21d0dddb1db/ios/Frame%20Processor/FrameProcessorRuntimeManager.mm#L96), not into the React-JS runtime. I don't know if it would also work if the native host functions are installed in the React-JS runtime, so I'll have to test that - but I believe Reanimated does it that way with their native functions (`measure`, ...).

It should work like this:

* Use reanimated from master branch, at least until my PR lands in a release
* Add the native host function to the babel.config globals, as mentioned in the VisionCamera docs. For MMKV get, this would be `mmkvGetString` (see [here](https://github.com/mrousavy/react-native-mmkv/blob/60caa204831fc59b51be5ea68d327f1604b4dfdc/ios/Mmkv.mm#L76) or [here](https://github.com/mrousavy/react-native-mmkv/blob/60caa204831fc59b51be5ea68d327f1604b4dfdc/src/index.ts#L22))
* Restart metro bundler for babel.config changes to take effect
* Call function from inside worklet, either use `mmkvGetString`, `global.mmkvGetString`, or `MMKV.getString` (imported). One of those should work.

**Comment by gillpeacegood** [comment]
Yes the above is pretty much exactly what I've done (I will give mmkv a quick go)

**Comment by gillpeacegood** [comment]
https://github.com/gillpeacegood/mmkv-test-async - crashes on async (but the function does get called, so not sure what was wrong with my other one)

**Comment by mrousavy** [comment]
Okay, thanks for the info! I'll try to find some time to investigate this soon.

**Comment by gursesl** [comment]
Pretty cool! I have a computationally heavy task of calculating loan amortization tables that can be offloaded to a portlet. So far I've been unable to call the host function, even with the instructions provided. Please keep us updated on any upcoming changes that can help.

Many thanks.
